src/main/java/org/opentripplanner/updater/vehicle_rental/VehicleRentalUpdater.java

```diff
@@ -133,63 +129,121 @@ public void setup(Graph graph) {
-                applyRegions(graph);
-            }
+            applyStations(graph);
+            applyRegions(graph);
+            service.setErrorsForNetwork(network, errors);
+            service.setSystemInformationDataForNetwork(network, systemInformationData);
         }
 
         private void applyStations(Graph graph) {
-            // Apply stations to graph
-            Set<VehicleRentalStation> stationSet = new HashSet<>();
-            Set<String> defaultNetworks = new HashSet<>(Arrays.asList(network));
-            LOG.info("Updating {} vehicle rental stations for network {}.", stations.size(), network);
-            /* add any new stations and update vehicle counts for existing stations */
-            for (VehicleRentalStation station : stations) {
-                service.addVehicleRentalStation(station);
-                stationSet.add(station);
-                VehicleRentalStationVertex vertex = verticesByStation.get(station);
-                if (vertex == null) {
-                    makeVertex(graph, station);
-                } else if (vertex.hasDifferentApproximatePosition(station)) {
-                    LOG.info("Vehicle rental {} has changed position, re-graphing", station);
-
-                    // First remove the old vertices and edges
-                    splitter.removeRentalStationVertexAndAssociatedSemiPermanentVerticesAndEdges(vertex);
-
-                    // then make a new vertices and edges
-                    makeVertex(graph, station);
-                } else {
-                    vertex.setVehiclesAvailable(station.vehiclesAvailable);
-                    vertex.setSpacesAvailable(station.spacesAvailable);
+            // check if any critical errors occurred
+            boolean feedWideError = false;
+            boolean allStationsError = false;
+            boolean allFloatingVehiclesError = false;
+            for (RentalUpdaterError error : errors) {
+                switch (error.severity) {
+                    case FEED_WIDE:
+                        feedWideError = true;
+                        break;
+                    case ALL_STATIONS:
+                        allStationsError = true;
+                        break;
+                    case ALL_FLOATING_VEHICLES:
+                        allFloatingVehiclesError = true;
+                        break;
                 }
             }
-            // Remove existing stations that were not present in the update
-            List<VehicleRentalStation> toRemove = new ArrayList<>();
+
+            Set<VehicleRentalStation> toRemove = new HashSet<>();
+            Set<VehicleRentalStation> stationsInUpdate = new HashSet<>();
+            LOG.info("Updating vehicle rental stations for network {}.", network);
+
+            // Apply stations to graph if a feed-wide error did not occur
+            if (!feedWideError) {
+                /* add any new stations that have fresh-enough data and update vehicle counts for existing stations */
```

Comment:
Change to `//` comment.

Reply:
fixed in 0b5b987b035c15264f45fc1eaf1e0cc5979fbe38